commerce_coordinator/apps/commercetools/tasks.py

```diff
@@ -70,32 +63,33 @@ def _log_info_and_release_lock(log_message):
 
     try:
         client = CommercetoolsAPIClient()
-        current_order_version = order_version
+        order = client.get_order_by_id(order_id)
+        current_order_version = order.version
 
-        cache_entry = cache.get(cache_key, None)
-
-        # TODO: Remove logging after testing on stage
-        if cache_entry:
-            logger.info(f'[CT-{tag}] Found cache entry for order version {cache_entry} for cache key {cache_key}')
-            current_order_version = cache_entry
-        else:
-            logger.info(f'[CT-{tag}] Cache entry not found for order version for cache key {cache_key}')
-
-        if self.request.retries > 0:
-            current_order_version = client.get_order_by_id(order_id).version
+        line_item = get_edx_line_item(order.line_items, line_item_id)
+        # from here we will always be transitioning from a 'Fulfillment Processing' state
+        line_item_state_id = client.get_state_by_key(TwoUKeys.PROCESSING_FULFILMENT_STATE).id
 
         updated_order = client.update_line_item_on_fulfillment(
             entitlement_uuid,
             order_id,
             current_order_version,
             line_item_id,
-            item_quantity,
-            from_state_id,
+            line_item.quantity,
+            line_item_state_id,
             to_state_key
         )
-
-        cache.set(key=cache_key, value=updated_order.version, timeout=TASK_LOCK_EXPIRE)
     except CommercetoolsError as err:
+        if (isinstance(err.errors, list) and any(
+            TwoUKeys.PROCESSING_FULFILMENT_STATE in e.get("message", "")
+            for e in err.errors
+        )):
+            _log_info_and_release_lock(
+                f'[CT-{tag}] Order {order_id} line item {line_item_id} already in state {to_state_key}'
+                f'{entitlement_info} Releasing lock.'
+            )
+            return None
+
```

Comment:
Could you please explain this case?

Reply:
@shafqatfarhan suggested we should not retry if line item is already processed. So we handled it here

Reply:
but the condition, will check if there is `PROCESSING_FULFILMENT_STATE` value in error message it will not retry. 

Can there be any other error containing this word other than we are concerned for?

Reply:
as discussed over call. please revert this.